### PR TITLE
Ensure docs are built on every PR

### PR DESCRIPTION
**Pull request by gj**

### CODE REVIEW

File: .github/workflows/docs-preview.yml [hunk 1]
```diff
@@ -61,9 +61,7 @@ jobs:
         with:
           hugo-version: '0.79.1'
       - name: Build Hugo docs
-        run: |
-          make build
-          HUGO_ENV=production hugo -b "https://docs-preview.oso.dev/"
+        run: make build
```
Comment (samscott89) [inline .github/workflows/docs-preview.yml:64]: Doesn't this already run on every PR?

Comment (gj) [reply .github/workflows/docs-preview.yml:64]: The YAML was off. Thanks for making me look twice at it.

Comment (samscott89) [reply .github/workflows/docs-preview.yml:72]: Ohhhh

File: .github/workflows/docs-preview.yml [hunk 2]
```diff
@@ -3,8 +3,8 @@ on:
   push:
     branches:
       - main
-    pull_request:
-      types: [opened, synchronize, reopened]
+  pull_request:
```
Comment (samscott89) [inline .github/workflows/docs-preview.yml:6]: oof. my bad

File: .github/workflows/docs-preview.yml [hunk 3]
```diff
@@ -16,6 +16,14 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-check-${{ hashFiles('**/Cargo.lock') }}
```
Comment (samscott89) [inline .github/workflows/docs-preview.yml:25]: This should have a different cache key. Otherwise, they race and if one does less work that gets cached.

Comment (samscott89) [reply .github/workflows/docs-preview.yml:25]: Although we do make python-build and make wasm-build in both so maybe its fine?

Comment (gj) [reply .github/workflows/docs-preview.yml:25]: Why would the one that does less work get cached? Wouldn't the tortoise win the race?

**Review by samscott89 — APPROVED** [review]

**Pull request merged**